providers/implementations/ciphers/ciphercommon_block.c

```diff
@@ -105,33 +106,32 @@ void ossl_cipher_padblock(unsigned char *buf, size_t *buflen, size_t blocksize)
         buf[i] = pad;
 }
 
+/*
+ * ossl_cipher_unpadblock removes the padding from the final block in constant time.
+ * Note that if the padding is invalid, ERR_raise is not called in order to preserve
+ * constant-timedness.
+ */
 int ossl_cipher_unpadblock(unsigned char *buf, size_t *buflen, size_t blocksize)
 {
     size_t pad, i;
     size_t len = *buflen;
+    int gp = 1;
 
-    if(len != blocksize) {
-        ERR_raise(ERR_LIB_PROV, ERR_R_INTERNAL_ERROR);
-        return 0;
-    }
+    gp &= constant_time_eq_s(len, blocksize);
 
     /*
      * The following assumes that the ciphertext has been authenticated.
      * Otherwise it provides a padding oracle.
      */
     pad = buf[blocksize - 1];
-    if (pad == 0 || pad > blocksize) {
-        ERR_raise(ERR_LIB_PROV, PROV_R_BAD_DECRYPT);
-        return 0;
-    }
+    gp &= ~(constant_time_is_zero_s(pad) | constant_time_lt_s(blocksize, pad));
+
     for (i = 0; i < pad; i++) {
-        if (buf[--len] != pad) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_BAD_DECRYPT);
-            return 0;
-        }
+        gp &= constant_time_eq_s(buf[--len], pad);
     }
```

Comment:
This does not look like it's constant time. The for loop still depends on `pad`. If you want to have it constant time, you need to make sure the loop is always over the same amount of bytes.

Reply:
It is constant time now